vaadin-cdi/src/main/java/com/vaadin/cdi/annotation/CdiComponent.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2000-2022 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.cdi.annotation;
+
+import javax.enterprise.inject.Stereotype;
+import java.lang.annotation.*;
+
+/**
+ * This is a {@code @Stereotype} to let the container scan and manage
+ * the instances of the annotated types.
+ * <p>
+ * Annotating the beans with this annotation is necessary when the
+ * {@code bean-discovery-mode="annotated"} is set in the beans.xml
+ * file (implicit bean archive), and in this case CDI can only manage
+ * and inject beans annotated with a "scope" type.
```

Comment:
> can only manage and inject beans annotated with a "scope" type.

I cannot understand this statement. Do you mean beans annotated with a "normal scope"? In this case, I would rephrase as `can only manage and inject beans annotated with a bean defining annotations`.

Reply:
That statement is coming from https://docs.oracle.com/javaee/7/tutorial/cdi-adv001.htm. But, I agree that rephrasing it to what you proposed can make it clearer.